mockk/common/src/main/kotlin/io/mockk/impl/recording/states/VerifyingState.kt

```diff
@@ -54,16 +54,13 @@ class VerifyingState(
     }
 
     private fun failIfNotPassed(outcome: VerificationResult, inverse: Boolean) {
-        if (inverse) {
-            when (outcome) {
-                is VerificationResult.OK -> {
-                    val callsReport = VerificationHelpers.formatCalls(outcome.verifiedCalls)
-                    throw AssertionError("Inverse verification failed.\n\nVerified calls:\n$callsReport")
-                }
+        when (outcome) {
+            is VerificationResult.OK -> if (inverse) {
+                val callsReport = VerificationHelpers.formatCalls(outcome.verifiedCalls)
+                throw AssertionError("Inverse verification failed.\n\nVerified calls:\n$callsReport")
             }
-        } else {
-            when (outcome) {
-                is VerificationResult.Failure -> throw AssertionError("Verification failed: ${outcome.message}")
+            is VerificationResult.Failure -> if (!inverse) {
+                throw AssertionError("Verification failed: ${outcome.message}")
```

Comment:
I did a little bit of refactoring here because having two 'when' statements seemed odd. I think it's clearer now.